src/ad-products/templates/hide-on-viewability.ts

```diff
@@ -0,0 +1,92 @@
+import { AdSlot, context, slotTweaker, utils } from '@wikia/ad-engine';
+import { CSS_TIMING_EASE_IN_CUBIC, FADE_IN_TIME, SLIDE_OUT_TIME } from './uap/constants';
+
+export interface HideOnViewabilityTemplateConfig {
+	additionalHideTime: number;
+	slideOutEdge: string;
+	hideAnimation: (adSlot: AdSlot, edge: string) => void;
+}
+
+export class HideOnViewability {
+	/**
+	 * Returns template name
+	 */
+	static getName(): string {
+		return 'hideOnViewability';
+	}
+
+	/**
+	 * Returns default template config
+	 */
+	static getDefaultConfig(): HideOnViewabilityTemplateConfig {
+		return {
+			additionalHideTime: 0,
+			slideOutEdge: 'bottom',
+			hideAnimation: (adSlot, edge) => {
+				const height = adSlot.getElement().offsetHeight;
+				const hideStyles = {
+					transition:
+						`${edge} ${SLIDE_OUT_TIME}ms ${CSS_TIMING_EASE_IN_CUBIC}, ` +
+						`opacity ${FADE_IN_TIME}ms ${CSS_TIMING_EASE_IN_CUBIC}`,
+					opacity: '0',
```

Comment:
```suggestion
					opacity: '0',
					[edge]: `-${height}px`
```

and L34 can be removed

Reply:
I didn't know that :o Changed!